templates/jupyterhub_config.py.j2

```diff
@@ -22,12 +22,47 @@ c.PAMAuthenticator.open_sessions = False
 # Listen on all interfaces, since hub should be reachable from spawned nodes
 c.JupyterHub.hub_ip = '0.0.0.0'
 
+# Assign Qhub Spawner
 class QHubHPCSpawner(SlurmSpawner):
     req_conda_environment_prefix = Unicode('',
         help="Conda environment prefix to launch jupyterlab"
     ).tag(config=True)
 
+c.JupyterHub.template_paths = []
+c.JupyterHub.extra_handlers = []
+
+{% if cdsdashboards.enabled %}
+from cdsdashboards.app import CDS_TEMPLATE_PATHS
+from cdsdashboards.hubextension import cds_extra_handlers
+from cdsdashboards.hubextension.spawners.variablemixin import VariableMixin, MetaVariableMixin
+
+c.VariableMixin.extra_presentation_launchers = {'qhub-bokeh': {
+       'cmd': ['/opt/conda/envs/jupyterlab/bin/jhsingle-native-proxy'],            
```

Comment:
path need to be templated with jinja to get correct. e.g. "{{ miniforge.home }}/envs/{{ jupyterhub.lab_environment }}/bin/jhsingle-native-proxy"

Reply:
Do you think it's possible to leave off the path and just have the command be `jhsingle-native-proxy`.  I'm thinking of the steps the user would need to change to make this work with some arbitrary environment.

Reply:
I think eventually we'll need tighter integration with https://github.com/ideonate/cdsdashboards/blob/20a0944a09d4e66880fa24faf61b08f3430e85fc/share/cdsdashboards/templates-common/editdashboard.html#L171 and https://github.com/ideonate/cdsdashboards/blob/28a237e24801b42f65d1ce1b7967edfc340be5ba/cdsdashboards/hubextension/main.py. But for now it doesn't hurt to hardcode. Lets open an issue to allow users to choose the conda environment for dashboarding. I'm not a fan of ambiguous paths if we can avoid it. Because it makes it harder to debug which environment that command is being run from.